File: app/src/main/java/com/syepro/app/commonjar/utils/ToastUtils.java
package com.syepro.app.commonjar.utils;

import android.content.Context;
import android.text.TextUtils;
import android.widget.Toast;

import com.syepro.app.base.BaseApplication;

/**
 * 吐司工具类
 */
public class ToastUtils {

    private static String URL_REG;

    static {
        URL_REG = "(http(s)?://)([\\w-]+\\.)+[\\w-]+(/[\\w- ./?%&#=]*)?";
    }

    public static void showToast(String msg) {
        showToast(BaseApplication.getInstance(), msg);
    }

    public static void showToast(Context ctx, String message) {
        if (TextUtils.isEmpty(message)) {
            message = "请求网络失败";
        }
        Toast.makeText(ctx, message, Toast.LENGTH_SHORT).show();
    }

    public static void showToast(String content, int gravity) {
        Toast toast = Toast.makeText(BaseApplication.getInstance(), content, Toast.LENGTH_SHORT);
        toast.setGravity(gravity, 0, 0);
        toast.show();
        showToast(content);
    }

    public static void showToast(int resId) {
        Toast.makeText(BaseApplication.getInstance(), resId, Toast.LENGTH_SHORT).show();

    }
}

File: kotlindemo/src/main/java/com/yulu/kotlindemo/utils/ToastUtils.kt
package com.yulu.kotlindemo.utils

import android.content.Context
import android.text.TextUtils
import android.widget.Toast
import com.yulu.kotlindemo.R

import com.yulu.kotlindemo.base.BaseApplication

/**
 * 吐司工具类
 */
object ToastUtils {

    private var URL_REG: String? = null

    init {
        URL_REG = "(http(s)?://)([\\w-]+\\.)+[\\w-]+(/[\\w- ./?%&#=]*)?"
    }

    fun showToast(msg: String) {
        showToast(BaseApplication.instance, msg)
    }

    fun showToast(ctx: Context, message: String) {
        var message = message
        if (TextUtils.isEmpty(message)) {
            message = ctx.getString(R.string.net_state_error)
        }
        Toast.makeText(ctx, message, Toast.LENGTH_SHORT).show()
    }

    fun showToast(content: String, gravity: Int) {
        val toast = Toast.makeText(BaseApplication.instance, content, Toast.LENGTH_SHORT)
        toast.setGravity(gravity, 0, 0)
        toast.show()
        showToast(content)
    }

    fun showToast(resId: Int) {
        Toast.makeText(BaseApplication.instance, resId, Toast.LENGTH_SHORT).show()
    }
}
